### PR TITLE
feat(SPV-778): adds mock responding with nills for QueryTransacion

### PR DESCRIPTION
**Pull request by jakubmkowalski**
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/go-broadcast-client/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/go-broadcast-client/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

**Review by Nazarii-4chain — APPROVED** [review]

File: broadcast/broadcast-client-mock/mock_client_builder.go [hunk 1]
```diff
@@ -40,6 +41,8 @@ func (cb *builder) WithMockArc(mockType MockType) *builder {
 		clientToReturn = mocks.NewArcClientMockFailure()
 	case MockTimeout:
 		clientToReturn = mocks.NewArcClientMockTimeout()
+	case MockNillQueryTransactionResponse:
```
Comment (wregulski) [inline broadcast/broadcast-client-mock/mock_client_builder.go:44]: Please make it a bit shorter, for example:
`MockNilQueryTxResp`

File: broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package mocks
+
+import (
+	"context"
+
+	broadcast_api "github.com/bitcoin-sv/go-broadcast-client/broadcast"
+	"github.com/bitcoin-sv/go-broadcast-client/broadcast/broadcast-client-mock/fixtures"
+)
+
+type ArcClientMockNilQueryTransactionResponse struct{}
```
Comment (wregulski) [inline broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go:10]: the same as above, please make it a bit shorter

**Review by wregulski — CHANGES_REQUESTED** [review]

File: broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+// GetFeeQuote returns a successful FeeQuote response.
+func (*ArcClientMockNilQueryTransactionResponse) GetFeeQuote(ctx context.Context) ([]*broadcast_api.FeeQuote, error) {
+	quotes := make([]*broadcast_api.FeeQuote, 0)
+	quotes = append(quotes, Fee1)
+	quotes = append(quotes, Fee2)
```
Comment (ac4ch) [inline broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go:16]: u could add this in one line `quotes := []*broadcast_api.FeeQuote{Fee1, Fee2}`

File: broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+	return quotes, nil
+}
+
+// GetPolicyQuote return a successful PolicyQuoteResponse.
+func (*ArcClientMockNilQueryTransactionResponse) GetPolicyQuote(ctx context.Context) ([]*broadcast_api.PolicyQuoteResponse, error) {
+	policies := make([]*broadcast_api.PolicyQuoteResponse, 0)
+	policies = append(policies, Policy1)
+	policies = append(policies, Policy2)
```
Comment (ac4ch) [inline broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go:25]: same as in above

File: broadcast/broadcast-client-mock/mock_client_builder.go (hunk 1 above)
Comment (jakubmkowalski) [reply broadcast/broadcast-client-mock/mock_client_builder.go]: Changed to shorter name

File: broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go (hunk 2 above)
Comment (jakubmkowalski) [reply broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go]: Changed to shorter name

File: broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go (hunk 3 above)
Comment (jakubmkowalski) [reply broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go]: Reduced to one line

File: broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go (hunk 4 above)
Comment (jakubmkowalski) [reply broadcast/internal/arc/mocks/arc_client_mock_success_nill_query_transaction_response.go]: Reduced to one line

**Review by wregulski — APPROVED** [review]

**Pull request merged**